src/mgr/mgr_commands.cc

```diff
@@ -14,7 +14,11 @@
 // enabled set (or upgrade script to run `disable <module`). At the time of
 // writing, the proposed always-on modules are being added within the same
 // release.
-const std::set<std::string> always_on_modules = {"crash"};
+const std::set<std::string> always_on_modules = {
+  "crash",
+  "status",
```

Comment:
Yeh, there'd be (at least) two ways to go about this. We could tag modules as always on like you suggest; i think this would require an update to the wire messages. The other option is to take advantage of the fact that list of always_on_modules` is already compiled into the monitor so we could use that. I think in both cases there are just a couple edge cases that can occur when upgrading depending on the ordering of monitor/manager restart. Any thoughts, I'm happy to put together a patch for this.

Reply:
@noahdesu no, what i suggested is to include the always-on modules in `MMgrBeacon::modules` also. and check the specified module in `mgr module` commands with the ones defined in `mgr_commands.cc`. i guess we are thinking about the very same fix.

> Any thoughts, I'm happy to put together a patch for this.

thanks in advance!

Reply:
@tchaikov had a chat with john this morning about this topic. flagging the always on module in the beacon is one way to handle this. we sort of converged on the idea that maybe the monitor should be the authority on always on modules, and control this by manipulating the manager map. i still need to work through the different scenarios to figure out if there are any issues.

Reply:
quote from the comment in the above code:

> always_on modules do not appear in the list of modules that can be enabled or disabled by the user.

@noahdesu the reason i suggested that we should include the "always-on" modules in `MMgrBeacon::modules` is listed in the "enabled_modules" 

see /a/kchai-2018-08-22_08:29:54-rados-wip-26994-distro-basic-smithi/2936591/teuthology.log . so from the user's perspective, it's always possible to disable an "enabled" module.

probably we can just follow @jcsp 's  suggestion to remove the test using `mgr module enable status`, as "selftest" is just what we need to exercise this functionality.